terraform/modules/bosh_vpc/endpoint_s3.tf

```diff
@@ -11,22 +11,31 @@ resource "aws_vpc_endpoint" "private-s3" {
         "Action": "s3:*",
         "Effect": "Allow",
         "Resource": "*",
-        "Principal": "*"      
-    }]
-}
-EOF
-
-}
-
-/* This needs to be in place for trusted s3 egress to work. Removing from policy ^^ due to ECR build issues with federalist
-        ,
+        "Principal": "*",
         "Condition": {
           "ForAllValues:StringEquals": {
             "aws:PrincipalAccount": ${jsonencode(local.policy_account_list)},
             "aws:ResourceAccount": ${jsonencode(local.policy_account_list)}
           }				
-        } 
-*/
+        }        
+    },
+    {
+      "Sid": "Access-to-ecr-buckets",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "s3:*",
+      "Resource": [
```

Comment:
Probably an ignorant question: but do we not want the same conditions to restrict only traffic from the expected accounts to these buckets?

Reply:
I dont think it would be needed, this pol statement should allow the ECR service itself to access its own bucket outside our account. The principle account condition might not apply to the service principal in this case, or it might block the service. And by the nature of its deployment, this policy only applies to traffic within the account its deployed in, unless we use a shared or peered vpc across accounts...

The way the condition filter works in the top statement is to check that the Iam user/role making the request is one that lives in an account we own, and that the Resource also lives in that same set of accounts.. This would prevent a client from creating a user in their own account and using those credentials in a CF app to access an S3 bucket in their own account under trusted egress. For that they still need to use public_egress..

Reply:
Sounds good. Though the `Principal: "*"` in this statement means it's not just ECR getting access to these buckets through the gateway, but in theory anything, correct?

Reply:
Looks like I might be able to use a condition with:
 "Condition": {
          "ForAllValues:StringEquals": {
            "aws:PrincipalServiceName": "ecr.amazonaws.com
    }
}

Reply:
oK, just checking

Reply:
However even though the policy allows access to those buckets, unless the ECR service has those buckets open to the world, a user wouldn't be able to access them. They most likely have bucket policies in place..

Reply:
I appreciate the review and questions, Thanks Mark!